components/list/README.md

```diff
@@ -163,6 +163,99 @@ The `d2l-list` is the container to create a styled list of items using `d2l-list
     - Image: max dimensions: `width: 216px` and `height: 120px` and has `20px margin` from the main content;
     - default break: `843px < x`  where `x` is the width of the component.
 
+### add-button Property
+
+The `add-button` attribute inserts the `d2l-button-add` component above and below each item in a list. Each nested list that wants to have this functionality will need to have this attribute on the nested `d2l-list` as well. Each list item needs to have its own `key` in order to be able to communicate where a new item should be positioned.
+
+The adding of a new item into the list is to be entirely handled by consumers through listening for the `d2l-list-add-button-click` event. A simple example scenario is below.
+
+<!-- docs: demo code display:block -->
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/list/list.js';
+  import '@brightspace-ui/core/components/list/list-controls.js';
+  import '@brightspace-ui/core/components/list/list-item.js';
+  import '@brightspace-ui/core/components/list/list-item-content.js';
+  import '@brightspace-ui/core/components/selection/selection-action.js';
+  import { css, html, LitElement } from 'lit';
+  import { getUniqueId } from '@brightspace-ui/core/helpers/uniqueId.js';
+
+  class ListDemoAddButton extends LitElement {
+
+    _handleListAddButtonClick(e) {
```

Comment:
Note: this would not pass linting since they are private methods and would normally be required to be placed after `render`. I'm guessing that these are placed first just to help draw attention to them in the docs...?

Reply:
Yup that's exactly it. I debated over that but figured it was worth it to put them first to call attention to them because the rest of the code doesn't matter as much. Open to the other way too though.